core/src/test/java/org/elasticsearch/monitor/jvm/JvmGcMonitorServiceTests.java

```diff
@@ -0,0 +1,119 @@
+        final TimeValue currentColletionTime = TimeValue.timeValueMillis(randomIntBetween(1, elapsedValue));
+        final ByteSizeValue lastHeapUsed = new ByteSizeValue(randomIntBetween(1, 1 << 10));
+        final ByteSizeValue currentHeapUsed = new ByteSizeValue(randomIntBetween(1, 1 << 10));
+        final ByteSizeValue maxHeapUsed = new ByteSizeValue(Math.max(lastHeapUsed.bytes(), currentHeapUsed.bytes()) + 1 << 10);
+        JvmGcMonitorService.logSlowGc(
+            logger,
+            threshold,
+            name,
+            seq,
+            elapsed,
+            totalCollectionCount,
+            currentCollectionCount,
+            totalCollectionTime,
+            currentColletionTime,
+            lastHeapUsed,
+            currentHeapUsed,
+            maxHeapUsed,
+            "asdf");
+        if (threshold == JvmGcMonitorService.JvmMonitor.Threshold.WARN) {
+            verify(logger).isWarnEnabled();
+            verify(logger).warn(
+                "[gc][{}][{}][{}] duration [{}], collections [{}]/[{}], total [{}]/[{}], memory [{}]->[{}]/[{}], all_pools {}",
+                name,
+                seq,
+                totalCollectionCount,
+                currentColletionTime,
+                currentCollectionCount,
+                elapsed,
+                currentColletionTime,
+                totalCollectionTime,
+                lastHeapUsed,
+                currentHeapUsed,
+                maxHeapUsed,
+                "asdf");
+        } else if (threshold == JvmGcMonitorService.JvmMonitor.Threshold.INFO) {
+            verify(logger).isInfoEnabled();
+            verify(logger).info(
+                "[gc][{}][{}][{}] duration [{}], collections [{}]/[{}], total [{}]/[{}], memory [{}]->[{}]/[{}], all_pools {}",
+                name,
+                seq,
+                totalCollectionCount,
+                currentColletionTime,
+                currentCollectionCount,
+                elapsed,
+                currentColletionTime,
+                totalCollectionTime,
+                lastHeapUsed,
+                currentHeapUsed,
+                maxHeapUsed,
+                "asdf");
+        } else if (threshold == JvmGcMonitorService.JvmMonitor.Threshold.DEBUG) {
+            verify(logger).isDebugEnabled();
+            verify(logger).debug(
+                "[gc][{}][{}][{}] duration [{}], collections [{}]/[{}], total [{}]/[{}], memory [{}]->[{}]/[{}], all_pools {}",
+                name,
+                seq,
+                totalCollectionCount,
+                currentColletionTime,
+                currentCollectionCount,
+                elapsed,
+                currentColletionTime,
+                totalCollectionTime,
+                lastHeapUsed,
+                currentHeapUsed,
+                maxHeapUsed,
+                "asdf");
+        } else {
+            fail("unexpected threshold [" + threshold + "]");
+        }
+        verifyNoMoreInteractions(logger);
```

Comment:
The mockito folks [advise](http://site.mockito.org/mockito/docs/current/org/mockito/Mockito.html#finding_redundant_invocations) against using this method too much and this isn't one of those times where I'd generally feel compelled to use it.